frontend/src/modules/Shares/components/ShareItemFilterModal.js

```diff
@@ -0,0 +1,386 @@
+          },
+          variant: 'success'
+        });
+        if (reloadItems) {
+          reloadItems();
+        }
+        if (onApply) {
+          onApply();
+        }
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
+    } catch (err) {
+      setStatus({ success: false });
+      setErrors({ submit: err.message });
+      setSubmitting(false);
+      dispatch({ type: SET_ERROR, error: err.message });
+    }
+  }
+
+  const remove = async (attachedDataFilterUri) => {
+    setRemoving(true);
+    const response = await client.mutate(
+      removeShareItemFilter({ attachedDataFilterUri: attachedDataFilterUri })
+    );
+    if (!response.errors) {
+      enqueueSnackbar('Removed data filters from table item', {
+        anchorOrigin: {
+          horizontal: 'right',
+          vertical: 'top'
+        },
+        variant: 'success'
+      });
+      if (reloadItems) {
+        reloadItems();
+      }
+      if (onApply) {
+        onApply();
+      }
+    } else {
+      dispatch({ type: SET_ERROR, error: response.errors[0].message });
+    }
+    setRemoving(false);
+  };
+
+  useEffect(() => {
+    if (client) {
+      if (itemDataFilter) {
+        setSelectionModel(itemDataFilter.dataFilterUris);
+        setSelectionModelNames(itemDataFilter.dataFilterNames);
+      }
+      fetchFilters().catch((e) =>
+        dispatch({ type: SET_ERROR, error: e.message })
+      );
+    }
+  }, [client, dispatch, fetchFilters]);
+
+  if (!item) {
+    return null;
+  }
+
+  if (viewOnly) {
+    return (
+      <Dialog maxWidth="lg" fullWidth onClose={onClose} open={open} {...other}>
+        <Box fullWidth sx={{ p: 3 }}>
+          <Typography
+            align="center"
+            color="textPrimary"
+            gutterBottom
+            variant="h4"
+          >
+            Data filters assigned to {item.itemName}
+          </Typography>
+          {loading ? (
+            <CircularProgress sx={{ mt: 1 }} size={20} />
+          ) : (
+            <DataGrid
+              fullWidth
+              autoHeight
+              scrollbarSize={50}
+              rowSpacingType="border"
+              rows={filters.filter((f) =>
+                itemDataFilter.dataFilterUris.includes(f.filterUri)
+              )}
+              getRowId={(filter) => filter.filterUri}
+              columns={columns}
+              pageSize={pageSize}
+              rowsPerPageOptions={[5, 10, 20]}
+              onPageSizeChange={(newPageSize) => setPageSize(newPageSize)}
+              loading={loading}
+            />
+          )}
+        </Box>
+      </Dialog>
+    );
+  }
+
+  return (
+    <Dialog maxWidth="lg" fullWidth onClose={onClose} open={open} {...other}>
+      <Box sx={{ p: 3 }}>
+        <Typography
+          align="center"
+          color="textPrimary"
+          gutterBottom
+          variant="h4"
+        >
+          Assign data filters to a table share item {item.itemName}
```

Comment:
nit: simplify title "Assign data filters to {item.itemName}"

Reply:
nit2: I would remove the select Data filters text, I think it is pretty clear the data filters need to be selected. As for the create data filter text I would turn it into a button, it would be great to compact it and place it next to the search bar.

nit3: this data grid in unstyled, we could add the boarders as commented for the TableView

![image](https://github.com/user-attachments/assets/6274020e-5ce5-436a-8a6a-724b95535986)

Reply:
For `nit2` - what search bar?

Reply:
![Screenshot 2024-08-07 at 10 49 28 AM](https://github.com/user-attachments/assets/fc243b64-d893-47c6-ba2f-761f4607533a)

is this more like what you envision?

Reply:
Perfect!